tools/importer.py

```diff
@@ -445,6 +448,123 @@ def execute(self, context):
+            #A dictionary to change the current model to MMD importer compatable temporarily
+            bonedict = {
+                "Chest":"UpperBody",
+                "Neck":"Neck",
+                "Head":"Head",
+                "Hips":"Center",
+                "Spine":"LowerBody",
+                
+                "Right wrist":"Wrist_R",
+                "Right elbow":"Elbow_R",
+                "Right arm":"Arm_R",
+                "Right shoulder":"Shoulder_R",
+                "Right leg":"Leg_R",
+                "Right knee":"Knee_R",
+                "Right ankle":"Ankle_R",
+                "Right toe":"Toe_R",
+                
+                
+                "Left wrist":"Wrist_L",
+                "Left elbow":"Elbow_L",
+                "Left arm":"Arm_L",
+                "Left shoulder":"Shoulder_L",
+                "Left leg":"Leg_L",
+                "Left knee":"Knee_L",
+                "Left ankle":"Ankle_L",
+                "Left toe":"Toe_L"
+                
+            }
+            
+            armature = Common.set_default_stage()
+            new_armature = armature.copy()
+            bpy.context.collection.objects.link(new_armature)
+            new_armature.data = armature.data.copy()
+            new_armature.name = "Cats MMD Rig Proxy"
+            new_armature.animation_data_clear()
+            
+            Common.unselect_all()
+            Common.set_active(new_armature)
+            Common.switch('OBJECT')
+            
+            for bone in new_armature.data.bones:
+                if bone.name in bonedict:
+                    bone.name = bonedict[bone.name]
+            
+            bpy.ops.mmd_tools.import_vmd(filepath=self.filepath,bone_mapper='RENAMED_BONES',use_underscore=True, dictionary='INTERNAL')
+            
+            #create animation for original if there isn't one.
+            if armature.animation_data == None :
+                armature.animation_data_create()
+            if armature.animation_data.action == None:
+                armature.animation_data.action = bpy.data.actions.new("MMD Animation")
+                
+                
+            #create animation for new if there isn't one.
+            if new_armature.animation_data == None :
+                new_armature.animation_data_create()
+            if new_armature.animation_data.action == None:
+                new_armature.animation_data.action = bpy.data.actions.new("EMPTY_SOURCE")
+            
+            active_obj = new_armature
+            ad = armature.animation_data
+            
+            #iterate through bones and translate them back, therefore blender API will change the animation to be correct.
+            for bone in new_armature.data.bones:
+                if bone.name in bonedict.values():
+                    bone.name = list(bonedict.keys())[list(bonedict.values()).index(bone.name)] #reverse name of bone from value in dictionary back to a key to change the animation.
```

Comment:
Replace this line by creating a dict outside the for loop that reverses the mapping of `bonedict`.
```reverse_bonedict = {v: k for k, v in bonedict.items()}```
Then, you can do
```bone.name = reverse_bonedict[bone.name]```

If you do this, it also means you won't even need to create a set like in my previous comment, because you can do
```if bone.name in reverse_bonedict:```